src/Compilers/CSharp/Portable/Binder/SwitchBinder.cs

```diff
@@ -205,7 +205,28 @@ private BoundExpression ConvertCaseExpression(TypeSymbol switchGoverningType, CS
 
             if (switchGoverningType.IsNullableType() && convertedCaseExpression.Kind == BoundKind.Conversion)
             {
-                constantValueOpt = ((BoundConversion)convertedCaseExpression).Operand.ConstantValue;
+                var operand = ((BoundConversion)convertedCaseExpression).Operand;
+
+                if (operand.ConstantValue.IsIntegral)
+                {
```

Comment:
Yes, you would need to do something like:
```C#
                    case (int)Foo.X:
                        return true;
```

I assume that works with this fix. As long as (int) is not getting dropped, it would not be a new case.

Just wanted to be sure we do not need to handle enums as well. They are often a missing case in fixes like this.